scipy/cluster/tests/test_vq.py

```diff
@@ -71,37 +75,53 @@
 
 
 class TestWhiten:
-    def test_whiten(self):
-        desired = np.array([[5.08738849, 2.97091878],
+    @array_api_compatible
+    def test_whiten(self, xp):
+        desired = xp.asarray([[5.08738849, 2.97091878],
                             [3.19909255, 0.69660580],
                             [4.51041982, 0.02640918],
                             [4.38567074, 0.95120889],
                             [2.32191480, 1.63195503]])
-        for tp in np.array, matrix:
+        arrays = [xp.asarray] if SCIPY_ARRAY_API else [np.asarray, matrix]
+        for tp in arrays:
             obs = tp([[0.98744510, 0.82766775],
                       [0.62093317, 0.19406729],
                       [0.87545741, 0.00735733],
                       [0.85124403, 0.26499712],
                       [0.45067590, 0.45464607]])
-            assert_allclose(whiten(obs), desired, rtol=1e-5)
+            if "cupy" in xp.__name__:
+                import cupy as cp
+                cp.testing.assert_allclose(whiten(obs), desired, rtol=1e-5)
```

Comment:
Yeah this would probably also help to have that for Pytorch. So makes sense to have a common function. But I suppose it would make sense to propose that to go in `array_api_compat`?